mmdet/models/losses/gfocal_loss.py

```diff
@@ -184,6 +188,24 @@ def forward(self,
                 calculate_loss_func = quality_focal_loss_with_prob
             else:
                 calculate_loss_func = quality_focal_loss
+
+            if isinstance(target, torch.Tensor):
+                # this means that target is already in One-Hot form.
+                assert pred.dim() == target.dim()
+                if target.dim() == 4:
```

Comment:
Should this method accept so many different types & shapes of input? Why not flatten the targets outside the loss function?

Reply:
This is because bce loss supports arbitrary dimensions. We hope to implement loss direct replacement